src/main/java/com/endava/webapp/service/DepartmentService.java

```diff
@@ -0,0 +1,39 @@
+package com.endava.webapp.service;
+
+import com.endava.webapp.dto.DepartmentDTO;
+import com.endava.webapp.entity.Department;
+import com.endava.webapp.mapper.DepartmentMapper;
+import com.endava.webapp.repository.DepartmentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class DepartmentService {
+
+    @Autowired
+    private DepartmentRepository departmentRepository;
+
+    @Autowired
+    private DepartmentMapper departmentMapper;
+
+    public List<DepartmentDTO> findAll() {
+        return departmentRepository.findAll()
+                .stream()
+                .map(departmentMapper::departmentToDepartmentDTO)
+                .collect(Collectors.toList());
+    }
+
+    public DepartmentDTO findById(long departmentId) {
+        Department department = departmentRepository.findById(departmentId).orElse(null);
```

Comment:
if nothing was found then threow by default Error. .orElseThrow() is ok

Reply:
Fixed.